make_queue/static/make_queue/js/stream.js

```diff
@@ -1,7 +1,7 @@
 function setupSocket($elem) {
     let chatSocket = new WebSocket(
         `wss://${window.location.host}/ws/stream/`
-        + $elem.attr("name").replace(/ /g, "-").replace(/ö/g, "o") + '/');
+        + $elem.attr("data-stream").replace(/ /g, "-").replace(/ö/g, "o") + '/');
```

Comment:
Since we now have a separate field for the stream name, the `replace()` calls can be removed. It would be better to do some validation in the machine form.

Reply:
It is there for backwards compatibility, so that the the `name` can be copied over to the `stream_name` in the database, before we manually change the websocket url

Reply:
Wouldn't it be better to set the `stream_name` in the migration so that they automatically are set to the sanitized version?

Reply:
Ah, I am not familiar with migrations, so I did not know that was a possibility. That seems like a better solution. I'll remove the `replace` calls.

Reply:
You can add this at the end of the `operations` list in the migrations:
```python
migrations.RunPython(set_stream_name, migrations.RunPython.noop),
```
and define the conversion function:
```python
def set_stream_name(apps, schema_editor):
    Machine = apps.get_model('make_queue', 'Machine')
    db_alias = schema_editor.connection.alias

    for machine in Machine.objects.using(db_alias).filter(machine_type__has_stream=True):
        machine.stream_name = machine.name.replace("ö", "o").replace(" ", "-")
        machine.save()
```

Reply:
Where do I define the `set_stream_name` function? When I put it at the end of the "0017_machine_stream_name" file, I get the error:

```
web_1  |   File "/web/make_queue/migrations/0017_machine_stream_name.py", line 18, in Migration
web_1  |     migrations.RunPython(set_stream_name, migrations.RunPython.noop),
web_1  | NameError: name 'set_stream_name' is not defined
```

Reply:
Moved it above the class, and now it works